### PR TITLE
Zahlenraten Main Feature

### PR DESCRIPTION
**Pull request by kittyevi**
PR: #1

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import random
+import sys
```
Comment (stefanluth) [inline main.py]: Unused import :)

File: main.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+user_name = input("Bitte geben Sie Ihren Benutzernamen ein: ")
+print(f"Hallo {user_name}! Ich denke an eine Zahl zwischen 1 und 20. Welche Zahl ist es?")
+
```
Comment (stefanluth) [inline main.py]: Put this in a function as well :)

File: main.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+def main_game():
+        random_number = random.randint(1, 20)
+        amount_of_tries = 1
+        guess = try_input(amount_of_tries)
+        while not test_input(random_number, guess, amount_of_tries):
+            guess = try_input(amount_of_tries)
+            amount_of_tries += 1
+            
+main_game()
```
Comment (stefanluth) [inline main.py]: Add a `if __name__ == '__main__'` here :)

File: main.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import random
+import sys
+
+user_name = input("Bitte geben Sie Ihren Benutzernamen ein: ")
+print(f"Hallo {user_name}! Ich denke an eine Zahl zwischen 1 und 20. Welche Zahl ist es?")
+
+def try_input(amount_of_tries):
+    while True:
+            try:
```
Comment (stefanluth) [inline main.py]: Indentation seems off :D

File: main.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+def try_input(amount_of_tries):
+    while True:
+            try: 
+                if amount_of_tries == 1:
+                    guess = int(input("Bitte geben Sie eine Zahl zwischen 1 und 20 ein: "))
+                else: 
+                    guess = int(input())
+
+                if (guess < 1 or guess > 20):
```
Comment (stefanluth) [inline main.py]:
```suggestion
                if 1 > guess > 20:
```

This _should_ work, but test it :D

File: main.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                    raise ValueError
+            except ValueError:
+                if amount_of_tries == 1:
+                    print("Eingabe unzulaessig.")
+                else: 
+                    print("Eingabe unzulaessig. Bitte geben Sie erneut eine Zahl zwischen 1 und 20 ein:  ", end="")
+                continue
```
Comment (stefanluth) [inline main.py]: This whole function is kind of over-engineered. Try to simplify :)

File: main.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        print(f"Wow. Punktlandung. :) Du hast {amount_of_tries} Versuche gebraucht.")
+
+        return True
+    
+
+def main_game():
+        random_number = random.randint(1, 20)
+        amount_of_tries = 1
+        guess = try_input(amount_of_tries)
+        while not test_input(random_number, guess, amount_of_tries):
```
Comment (stefanluth) [inline main.py]: The function name does not suggest a boolean return. Maybe `is_valid_input` :D

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: main.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+def try_input(amount_of_tries):
+    while True:
+        try: 
+            if amount_of_tries == 1:
+                guess = int(input("Bitte geben Sie eine Zahl zwischen 1 und 20 ein: "))
+            else: 
+                guess = int(input())
+
+            if (1 > guess > 20):
```
Comment (DevBastiS) [inline main.py]: 1 and 20 are magic numbers here, move them to constants with speaking name.

File: main.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                raise ValueError
+        except ValueError:
+            if amount_of_tries == 1:
+                print("Eingabe unzulaessig.")
+            else: 
+                print("Eingabe unzulaessig. Bitte geben Sie erneut eine Zahl zwischen 1 und 20 ein:  ", end="")
```
Comment (DevBastiS) [inline main.py]: Is this if case necessary?

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: main.py (hunk 5 above)
Comment (kittyevi) [reply main.py]: Doesn't work unfortunately

File: main.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
+import random
+
+min_random_number_limit = 5
+max_random_number_limit = 19
```
Comment (stefanluth) [inline main.py]:
```suggestion
MIN_NUM = 5
MAX_NUM = 19
```

Constants are UPPER_CASE :)

File: main.py [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    while True:
+        try: 
+            guess = int(input(f"Bitte geben Sie eine Zahl zwischen {min_random_number_limit} und {max_random_number_limit} ein: "))
+            if (min_random_number_limit > guess or guess > max_random_number_limit):
+                raise ValueError
+        except ValueError:
+            print("Eingabe unzulaessig. ", end="")
+            continue
+        else:
+            return guess
```
Comment (stefanluth) [inline main.py]:
```suggestion
        try: 
            guess = int(input(f"Bitte geben Sie eine Zahl zwischen {min_random_number_limit} und {max_random_number_limit} ein: "))
            if (min_random_number_limit > guess or guess > max_random_number_limit):
                raise ValueError
            return guess
        except ValueError:
            print("Eingabe unzulaessig. ", end="")
            continue
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: main.py [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+def welcome_username_prompt(): 
+    user_name = input("Bitte geben Sie Ihren Benutzernamen ein: ")
+    print(f"Hallo {user_name}! Ich denke an eine Zahl zwischen {MIN_NUM} und {MAX_NUM}. Welche Zahl ist es?")
+
+def try_input():
+    while True:
+        try: 
+            guess = int(input(f"Bitte geben Sie eine Zahl zwischen {MIN_NUM} und {MAX_NUM} ein: "))
+            if (MIN_NUM > guess or guess > MAX_NUM):
+                raise ValueError
```
Comment (gazer42) [inline main.py:23]:
```suggestion
                raise ValueError
                
```

File: main.py [hunk 13]
```diff
@@ -0,0 +1,47 @@
+import random
```
Comment (gazer42) [inline main.py]: since you only need the randint func, only import this func

**Review by gazer42 — CHANGES_REQUESTED** [review]

**Review by stefanluth — APPROVED** [review]

**Review by gazer42 — APPROVED** [review]

**Review by DevBastiS — APPROVED** [review]

**Pull request merged**